backend/models/plan.py

```diff
@@ -1,4 +1,118 @@
+        self.admin: User = admin  # aka the creator
+        self.date: datetime = date
+        self.location: str = location
+        self.participants: list[User] = [admin] if participants is None else participants
+
+    def json(self):
+        return {
+            'id': self.uid,
+            'name': self.name,
+            'description': self.description,
+            'date': self.date.strftime('%d/%m/%Y'),
+            'location': self.location,
+            #'admin': self.admin.json(),
+            'admin': self.admin,
+            'participants': self.participants
+            #'participants': [p.json() for p in self.participants]
+        }
+
+    @classmethod
+    def from_dict(cls, d: dict, include_uid=True):
+        try:
+            if include_uid:
+                return Plan(d[u'name'], d[u'description'], d[u'date'], d[u'location'], d[u'admin'], d[u'participants'],
+                            d[u'uid'])
+            return Plan(d[u'name'], d[u'description'], d[u'date'], d[u'location'], d[u'admin'], d[u'participants'],
+                        None)
+        except Exception as e:
+            raise PlanCreationError(d) from e
+
+    def add_plan(self):
+        try:
+            update_time, us_ref = db.collection(u'plans').add({
+                u'name': self.name,
+                u'description': self.description,
+                u'date': self.date,
+                u'location': self.location,
+                u'admin': self.admin.uuid if isinstance(self.admin, User) else self.admin,
+                u'participants': [p.uuid for p in self.participants] if isinstance(self.participants[0],
+                                                                                   User) else self.participants
+            })
+            self.uid = us_ref.id
+        except Exception as e:
+            raise PlanDBAddingError() from e
+
+    def update_plan(self):
+        try:
+            p_ref = db.collection(u'plans').document(self.uid)
+            p_ref.set({
+                u'name': self.name,
+                u'description': self.description,
+                u'date': self.date,
+                u'location': self.location,
+                u'admin': self.admin.uuid if isinstance(self.admin, User) else self.admin,
+                u'participants': [p.uuid for p in self.participants] if isinstance(self.participants[0],
+                                                                                   User) else self.participants
+            }, merge=True)
+            self.uid = p_ref.id
+        except Exception as e:
+            raise PlanDBAddingError() from e
+    @classmethod
+    def get_plan_by_id(cls, id: str):
+        p = db.collection(u'plans').document(id).get().to_dict()
+        if p is None:
+            raise PlanNotFoundError(id)
+        p['uid'] = id
+        try:
+            return Plan.from_dict(p)
+        except Exception as e:
+            raise PlanCreationError(p) from e
+
+    """
```

Comment:
Why is commented?

Reply:
currently working on get_user_plans(). Commented to clarify thay is not finished yet
add/remove participant will be finished in the appropiate PR

but yeah i can remove the comments and leave it as code

Reply:
it is not necessary